packages/fx-core/src/plugins/solution/fx-solution/commonQuestions.ts

```diff
@@ -208,6 +219,25 @@ export async function askResourceGroupInfo(
   }
 }
 
+async function getLocations(
+  ctx: SolutionContext,
+  rmClient: ResourceManagementClient
+): Promise<string[] | undefined> {
+  const credential = await ctx.azureAccountProvider!.getAccountCredentialAsync();
+  const subscriptionClient = new SubscriptionClient(credential as any);
+  const askSubRes = await ctx.azureAccountProvider!.getSelectedSubscription(true);
```

Comment:
shall we check if the result of `askSubRes` is undefined here? for example, the selection of subscription is cancelled by user.

Reply:
No, ```getSelectedSubscription``` already handles this part.

Reply:
From the method description https://github.com/OfficeDev/TeamsFx/blob/dev/packages/api/src/utils/login.ts#L83, subscriptionId will be empty string if user cancel the selection. I don't think the case has been handled correctly.

Reply:
When ```triggerUI == true```, means users have to login and select subscription. Otherwise this method will throw user cancel error.

Reply:
Offline discussed with @xiaolang124 , the user error caused by cancel operation will be handled and won't be exposed to user.